### ISSUE TITLE
Cannot run lint

### ISSUE THREAD

**Issue by paul121** [farmOS/farmOS-map#102]
I tried to `npm run build` and it's failing at the first command, `npm run lint`:

`Error: Cannot find module 'eslint-plugin-prettier'`

This might just be an issue in my dev environment but I'm pretty sure it used to work. I cloned the repo to a new directory and ran `npm install` with the most recent changes. I tried installing the modules as needed, but kept getting errors:

`npm install --save-dev eslint-plugin-prettier`

then...

`Error: Cannot find module 'eslint-config-prettier'`

so....

`npm install --save-dev eslint-config-prettier`

then....

`Error: Cannot find module 'prettier'`

so...

`npm install --save-dev prettier`

then it worked.... but it seems like it tries to use a new linting config, 455 errors:
```
✖ 455 problems (455 errors, 0 warnings)
  455 errors and 0 warnings potentially fixable with the `--fix` option.
```

Should my naive solution have worked? Or is something off?

Didn't look much further, but I noticed we're currently on eslint 5.16 and the latest version is 7.19: https://github.com/eslint/eslint/releases

**Comment by mstenta** [comment]
I can't reproduce this. I also tried cloning a fresh copy, running `npm install` and then `npm run lint` and it worked as expected.

@paul121 can you try a fresh install again (`git clone` to a new directory), then run `npm install`, followed by `npm run lint` and see if it works?

Closing this, but reopen if you hit the same issue. Seems like a dev environment issue, but happy to help debug.

**Comment by mstenta** [comment]
Might be worth checking the node/npm version you have?

`npm --version` gives me: `7.7.6`

**Comment by paul121** [comment]
So I was on `6.14.5`. Just updated to `7.8.0` and cloned into a new directory but having the same error as described above.

**Comment by mstenta** [comment]
I asked @jgaehring to test this, and he said it's working fine for him too.

Also notably: we use `npm run build` in our GitHub Action for packaging releases, and that just worked fine for the v1.4.2 release.

So it does seem to be a local environment issue.

FWIW, I run `npm` within a Docker container (partly to avoid installing things, but also maybe to help with reproducibility issues like this`). Here is the bash alias I have:

`alias npm='sudo docker run -it --rm -v /home/mstenta/.npm:/home/node/.npm -v "${PWD}":/usr/src/app -w /usr/src/app -u $UID -p 8080:8080 node npm'`

Note that the `-v /home/mstenta/.npm:/home/node/.npm` mounts MY `~/.npm` folder into the container, so you will either need to update or remove that. I think you might actually want to remove it in this case just to keep that variable out of the test... it could be that something is not right in your local `~/.npm` config (just a guess).

**Comment by jgaehring** [comment]
I wonder, @paul121, do you have Prettier installed globally? I think you can test with:

```bash
npm list -g prettier
```

FWIW, I use [nvm](https://github.com/nvm-sh/nvm) to manage my Node versions in local development. Sort of like virtualenv for Node.

**Comment by paul121** [comment]
> FWIW, I run npm within a Docker container

This worked for me! I took out the volume mount for the `.npm` folder - also tried resetting this in my local environment, but it didn't fix it.

> do you have Prettier installed globally? I think you can test with:

Ah interesting... looks like I did not. I tried install via `npm install -g prettier` and it seemed to install. But I still have the original issue.. Hadn't heard of `nvm` though, thanks for sharing. That sounds like the ultimate solution. But until I need the complexity I'm going to stick with the docker approach for now!

Thanks @mstenta @jgaehring :1st_place_medal:
